deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl

```diff
@@ -635,6 +635,19 @@ check_credentials(Username, Password, SslLoginName, PeerIp) ->
             {ok, {UserBin, PassBin}}
     end.
 
+%% Extract client_id from the certificate provided it was configured to do so and 
+%% it is possible to extract it else returns the client_id passed as parameter
+-spec extract_client_id_from_certificate(client_id(), rabbit_net:socket()) -> client_id().
+extract_client_id_from_certificate(Client0, Socket) ->
+    case extract_ssl_cert_client_id_settings() of 
+        none -> Client0;
+        SslClientIdSettings -> 
+            case ssl_client_id(Socket, SslClientIdSettings) of 
+                none -> Client0;
+                V -> V 
```

Comment:
I think we should enforce that the MQTT client ID as sent in the CONNECT packet matches the MQTT client ID as extracted from the client certificate because otherwise 2 different client IDs might be defined which is confusing on multiple levels.

Reply:
do you suggest to maybe close the connection with an error ?

Reply:
yes

Reply:
the client may choose to send a blank client_id via the protocol because it is carried in the certificate. Considering that, if the client_id is not blank, it must match the client_id found in the certificate.

Reply:
As per our conversation, both client_id(s) have to be the same otherwise the connection is refused